csrc/quantization/utils.cuh

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+/**
+ * Quantization utilities including:
+ *   Definition of qtypes.
+ *   Adjusted maximum values for qtypes.
+ *   Minimum scaling factors for qtypes.
+ */
+
+#include <cmath>
+#include <torch/types.h>
+
+#ifndef USE_ROCM
+  #include <c10/util/Float8_e4m3fn.h>
+  #define MAYBE_HOST_DEVICE C10_HOST_DEVICE
+#else
+  #include <ATen/hip/HIPContext.h>
+  #include <c10/util/Float8_e4m3fn.h>
+  #include <c10/util/Float8_e4m3fnuz.h>
+  // ROCm doesn't seem to need C10_HOST_DEVICE for static constexpr
+  #define MAYBE_HOST_DEVICE
+#endif
+
+template <typename T>
+struct is_quant_type {
```

Comment:
If you want to keep these, you can simplify them and inherit from `std::false_type`/`std::true_type`